cpp/src/arrow/array/builder_base.h

```diff
@@ -101,11 +101,6 @@ class ARROW_EXPORT ArrayBuilder {
   virtual Status AppendNull() = 0;
   virtual Status AppendNulls(int64_t length) = 0;
 
-  /// For cases where raw data was memcpy'd into the internal buffers, allows us
-  /// to advance the length of the builder. It is your responsibility to use
-  /// this function responsibly.
-  Status Advance(int64_t elements);
```

Comment:
I don't know, perhaps some users rely on this public API? @xhochy @wesm what do you think?

Reply:
reverted